exercises/CI/introduction_to_ci/README.md

```diff
@@ -0,0 +1,154 @@
+
+
+
+
+
+
+## Introduction
+Development teams working on a common codebase are **continuously** making changes to that common code. In order for a team member to know that their changes are compatible, **continuous** checking of the code base should be happening too. I.e. every time some one makes a change, the code base should be tested to ensure it is continuing to work as intended. The idea is that the more frequently changes are integrated, the smaller those changes are likely to be and therefore the lower the risk is that they will break something. This is where the practice of Continuous Integration (CI) comes in.
+
+> Continuous Integration is a software development practice where members of a team integrate their work frequently, usually each person integrates at least daily - leading to multiple integrations per day. Each integration is verified by an automated build (including test) to detect integration errors as quickly as possible. Many teams find that this approach leads to significantly reduced integration problems and allows a team to develop cohesive software more rapidly.
+- Martin Fowler: [https://martinfowler.com/articles/continuousIntegration.html](https://martinfowler.com/articles/continuousIntegration.html)
+
+Traditionally, large systems are very difficult to change and time consuming to verify. This results in new releases only being made a few times a year. These release are usually high risk, high anxiety, events due to their manual and error prone nature. CI is an automated process, this means, with careful management, that as the code base grows the time taken verify the code base should remain short (ideally just a few minutes). This has the knock on effect that new releases can be prepared and released frequently. Frequent, high quality, releases benefit the business and take the pressure off of the team.
+
+### Continuous Integration Servers
+Continuous integration servers are essentially glorified schedulers. They sit looking at your Version Control System (VCS) and when they see that a change has been introduced they do what ever they have been configured to do. E.g. to compile the new version of the code base and run a set of tests to see if it still works. Different CI servers may have different user interfaces, however they essentially all do the same thing and have a very common feature set. Therefore learn how to use one CI server and you should find that switching to another is a pretty easy process.
+
+## Exercise Learning Objectives
+- Introduction Concourse CI
+
+## Exercise
+**Note:** Before going any further do the following:
+- `cd YOUR_CLONE_OF_THIS REPO`
+- `source ./bin/env`
+- `cd `
+
+In this exercise you will be working using a CI server called Concourse. We've done the hard bit and have provided you with your very own Concourse installation. We've also provided you with a GIT server for your CI server to talk to. To start everything up, run: cic up 
```

Comment:
The `cic up` at the end of this line is not appearing as a command (missing backticks?)